tutorials/pubsub-spring-dedup-messages/index.md

```diff
@@ -0,0 +1,284 @@
+
+1. Set `GOOGLE_APPLICATION_CREDENTIALS` to your service account key file.
+
+   ```shell script
+   export GOOGLE_APPLICATION_CREDENTIALS=path/to/your/credentials.json
+   ```
+   If you have not done it, set your `PROJECT_NAME` environment variable to your GCP project.
+   ```shell script
+   export PROJECT_NAME=$(gcloud config get-value project)
+   ```
+
+1. Clone this repository and navigate to the sample code:
+
+   ```shell script
+   git clone https://github.com.GoogleCloudPlatform/pubsub-spring-dedup-messages.git
+   cd community/tutorials/pubsub-spring-dedup-messages
+   ```
+
+## Bind Pub/Sub to Your Spring Boot Application
+
+Spring Cloud Stream makes use of [binders] to send data to a Pub/Sub topic and receive data from a Pub/Sub subscription.
+
+Your application can send data from an internal queue to a Pub/Sub topic if you provide both of the following:
+1. a Spring Cloud Stream source
+1. a Pub/Sub topic
+
+### Specify a Spring Cloud Stream source
+Spring can recognize a Spring Cloud Stream source as a Supplier bean.
+
+In [App.java](pubsub-spring/src/main/java/com/google/example/App.java):
+
+[embedmd]:# (pubsub-spring/src/main/java/com/google/example/App.java java /  \/\/ The Supplier Bean/ /}/)
+```java
+  // The Supplier Bean makes the function a valid Spring Cloud Stream source. It
+  // sends messages to a Pub/Sub topic configured with the binding name
+  // `sendMessagesForDeduplication-out-0` in application.properties.
+  @Bean
+  Supplier<Flux<Message<String>>> sendMessagesForDeduplication(
+    final EmitterProcessor<Message<String>> frontEndListener) {
+    return () -> frontEndListener;
+  }
+```
+
+### Specify a Pub/Sub topic for the source
+To specify a Pub/Sub topic for the source, you can assign a topic name to the output binding destination of your source in `application.properties`. Here, the topic name is `topicToDataflow`. The output binding destination is indicated by `-out-` in the binding name `sendMessagesForDeduplication-out-0` and binding property `destination`. Spring takes input to the function `sendMessagesForDeduplication` and sends it to the Pub/Sub topic `topicToDataflow`. Spring will create this topic if it does not exist. For more information, see [Binding and Binding Names].
+
+In [application.properties](pubsub-spring/src/main/resources/application.properties):
+
+[embedmd]:# (pubsub-spring/src/main/resources/application.properties /.*Data going/ /=topicToDataflow/)
+```properties
+# Data going to Pub/Sub from a Spring Cloud Stream source defined in the
+# Supplier bean `sendMessagesForDeduplication`. The application will
+# create the Pub/Sub topic `topicToDataflow` if it does not exist.
+spring.cloud.stream.bindings.sendMessagesForDeduplication-out-0.destination=topicToDataflow
+```
+
+Similarly, your application can receive data from a Pub/Sub subscription if you specify both of the following: 
+1. a Spring Cloud Stream sink
+1. a pair of Pub/Sub topic and subscription
```

Comment:
I rephrased the paragraph below to say "a destination name and an optional group name". 

I also rephrased the sentence here to avoid misleading information.